Automattic/wp-calypso
# Migration: Skip site creation when siteId/siteSlug are available.

Closes #93553

## Proposed Changes
* Skip site creation steps when the siteId/siteSlug are available
* Remove necessary (platform) query params between step changes
* Rename inexpressive  query param `next`  to `importer`
* Reorganize test suite. 



## Why are these changes being made?
* We are going to have cases where the flow will be started with an existent site and it also affects the ticket #93308 (not implemented it);

## Testing Instructions
* Access `/setup/migration` 
*  Select any option
* Check a site was created (you can see a new siteId and siteSlug)
* Get the create siteId and siteSlug from the URL 
* Access the flow again but now passing the create site `/setup/migration?siteId=[SITE_ID]&siteSlug=[YOUR_SITE_SLUG]`
* Check you have been directly to the importer or checkout page using the you defined.


## Pre-merge Checklist
